android/src/main/java/com/polidea/flutter_ble_lib/FlutterBleLibPlugin.java

```diff
@@ -76,4 +90,30 @@ private void destroyClient(Result result) {
         result.success(null);
     }
 
+    private void startDeviceScan(@NonNull MethodCall call, Result result) {
+        bleAdapter.startDeviceScan(call.<String[]>argument(ArgumentKey.RESTORE_STATE_IDENTIFIER),
+                call.<Integer>argument(ArgumentKey.SCAN_MODE),
+                call.<Integer>argument(ArgumentKey.CALLBACK_TYPE),
+                new OnEventCallback<ScanResult>() {
+                    @Override
+                    public void onEvent(ScanResult data) {
+                        try {
+                            scanningStreamHandler.onScanResult(data);
+                        } catch (JSONException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                }, new OnErrorCallback() {
+                    @Override
+                    public void onError(BleError error) {
+                        Log.d("scanning_error", error.toString());
```

Comment:
Shouldn't the handler informed of error?

Reply:
`log.e` ?

Reply:
I will add error handling in another PR. We will send all errors from here straight to the scanningStreamHandler and then we will deserialise it on flutter side.